src/test/java/api/requests/RequestsAPICreationTests.java

```diff
@@ -1625,6 +1627,104 @@ private void recallItem(ItemResource item, List<UserResource> users) {
+    itemsFixture.basedUponSmallAngryPlanet(holdingBuilder -> holdingBuilder,
+      instanceBuilder -> instanceBuilder.withId(instanceId), ItemBuilder::inProcess);
+    itemsFixture.basedUponSmallAngryPlanet(holdingBuilder -> holdingBuilder,
+      instanceBuilder -> instanceBuilder.withId(instanceId), ItemBuilder::missing);
+
+    Response response = requestsFixture.attemptPlaceHoldOrRecallTLR(instanceId,
+      usersFixture.charlotte(), RECALL);
+
+    assertThat(response.getStatusCode(), CoreMatchers.is(422));
+    assertThat(response.getJson(), hasErrorWith(
+      hasMessage("Request has no loan or recallable item")));
+  }
+
+  @Test
+  void tlrRecallShouldPickOneFromAllowedStatusItemsIfLoanNotExistsAndInstanceHasAllowedItems() {
+    configurationsFixture.enableTlrFeature();
+    UUID instanceId = instancesFixture.basedUponDunkirk().getId();
+    IndividualResource defaultWithHoldings = holdingsFixture.defaultWithHoldings(instanceId);
+    itemsClient.create(new ItemBuilder()
+      .forHolding(defaultWithHoldings.getId())
+      .withMaterialType(UUID.randomUUID())
+      .withPermanentLoanType(UUID.randomUUID())
+      .withPermanentLoanType(loanTypesFixture.canCirculate().getId())
+      .withMaterialType(materialTypesFixture.book().getId())
+      .withPermanentLocation(locationsFixture.mainFloor())
+      .create());
+    requestsClient.create(buildPageTitleLevelRequest(usersFixture.jessica().getId(),
+      servicePointsFixture.cd1().getId(), instanceId));
+    itemsFixture.basedUponDunkirk(holdingBuilder -> holdingBuilder,
+      instanceBuilder -> instanceBuilder.withId(instanceId), ItemBuilder::awaitingDelivery);
+    itemsFixture.basedUponDunkirk(holdingBuilder -> holdingBuilder,
+      instanceBuilder -> instanceBuilder.withId(instanceId), ItemBuilder::awaitingPickup);
+
+    itemsFixture.basedUponDunkirk(holdingBuilder -> holdingBuilder,
+      instanceBuilder -> instanceBuilder.withId(instanceId), ItemBuilder::missing);
+
+    Response response = requestsFixture.attemptPlaceHoldOrRecallTLR(instanceId,
+      usersFixture.charlotte(), RECALL);
+
+    assertThat(response.getStatusCode(), CoreMatchers.is(201));
+    assertThat(response.getJson().getString("requestType"), is(RequestType.RECALL.getValue()));
+    assertTrue(Stream.of(AWAITING_PICKUP.getValue(), AWAITING_DELIVERY.getValue(),
+      PAGED.getValue()).anyMatch(status -> status.equals(response.getJson().getJsonObject("item")
+      .getString("status"))));
+    assertThat(response.getJson().getString("requestType"), is("Recall"));
+    assertThat(response.getJson().getString("requestLevel"), is("Title"));
+    assertThat(response.getJson().getString("instanceId"), is(instanceId));
+    assertThat(response.getJson().getString("fulfilmentPreference"), is("Hold Shelf"));
+    assertThat(response.getJson().getString("status"), is("Open - Not yet filled"));
+  }
+
+  @Test
+  void tlrRecallShouldPickItemWithAllowedStatusIfLoanNotExistsAndInstanceHasOtherItems() {
```

Comment:
Can we replace all your recall tests with one test where we:
- create items with every possible item status
- create recall requests until all items with recallable statuses are recalled (and verify every created request)
- verify that recall request fails when we are out of recallable items

Reply:
Tests are refactored as per agreement, thanks.